karyon3-core/src/main/java/com/netflix/karyon/KeyMatchers.java

```diff
@@ -0,0 +1,51 @@
+package com.netflix.karyon;
+
+import java.lang.annotation.Annotation;
+
+import com.google.inject.Key;
+import com.google.inject.matcher.AbstractMatcher;
+import com.google.inject.matcher.Matcher;
+
+/**
+ * Utility class for creating matchers for Key<?>
+ */
+public final class KeyMatchers {
```

Comment:
This is a utility class for creating the most common types of matchers where the most common use case is to match a specific type or having an annotation.  The whole processes of matching is done as part of the injector bootstrapping so injection doesn't make sense here.